### PR TITLE
Fix published POM to depend on correct version of htsjdk

### PR DESCRIPTION
**Pull request by karenfeng**
## What changes are proposed in this pull request?

The VCF piper has been broken in recently-released Glow Maven artifacts.

This was difficult to pinpoint as we threw away any encountered exceptions; now we re-throw any exceptions.

The problem boiled down to inconsistency in classpath resolution, as `dependencyOverrides` is not respected in `pom`s; we therefore unintentionally pull in an old version of the HTSJDK. To resolve this, we set the overrides as direct dependencies.

This PR also folds in a variety of smaller changes:
- Removes ADAM as a test dependency as it is not technically required
- Cleans up the Glow tests' Spark configs to match the recommended user environment
- Adds documentation to recommend setting the BGZF compression codec config
- Checks for approximate (rather than exact) row equality in the docs to allow for floating-point error

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests

Tested the VCF piper on the released Maven artifact, as well as a local Pyspark shell.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -7,14 +7,7 @@
...
         .master("local[2]") \
-        .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer") \
-        .config("spark.driver.maxResultSize", "0") \
-        .config("spark.kryo.registrator", "org.broadinstitute.hellbender.engine.spark.GATKRegistrator") \
-        .config("spark.kryoserializer.buffer.max", "2047m") \
-        .config("spark.kryo.registrationRequired", "false") \
-        .config(
-            "spark.hadoop.io.compression.codecs",
-            "org.seqdoop.hadoop_bam.util.BGZFCodec,org.seqdoop.hadoop_bam.util.BGZFEnhancedGzipCodec") \
+        .config("spark.hadoop.io.compression.codecs", "io.projectglow.sql.util.BGZFCodec") \
```
Comment (henrydavidge) [inline conftest.py]: Nice change

File: docs/source/tertiary/regression-tests.rst [hunk 2]
```diff
@@ -136,7 +136,7 @@ Example
 
 .. invisible-code-block: python
 
-   assert rows_equal(log_reg_df.head(), Row(contigName='20', start=10000053, names=[], beta=-0.04909334516505058, oddsRatio=0.9520922523419953, waldConfidenceInterval=[0.5523036168612923, 1.6412705426792646], pValue=0.8161087491239676))
+   assert rows_equal(log_reg_df.head(), Row(contigName='20', start=10000053, names=[], beta=-0.04909334516505058, oddsRatio=0.9520922533449219, waldConfidenceInterval=[0.5523036174500618, 1.6412705443874311], pValue=0.8161087511209355))
```
Comment (henrydavidge) [inline docs/source/tertiary/regression-tests.rst]: Why is this necessary?

File: core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala [hunk 3]
```diff
@@ -191,6 +192,8 @@ class PipeIterator(
     try {
       inputFormatter.init(stream)
       input.foreach(inputFormatter.write)
+    } catch {
+      case e: Throwable => logger.warn(e.toString)
```
Comment (henrydavidge) [inline core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala]: You still want to throw the exception. Also, prefer `NonFatal` to `Throwable`

**Comment by henrydavidge** [comment]
Also, the fact that this was so hard to debug means that we are incorrectly swallowing an exception somewhere. Is that fixed already in this PR?

File: core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala (hunk 3 above)
Comment (karenfeng) [reply core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala]: Good call - I wrapped this in `WithUtils.withCloseable`

File: docs/source/tertiary/regression-tests.rst (hunk 2 above)
Comment (karenfeng) [reply docs/source/tertiary/regression-tests.rst]: I made our tests more permissive to floating-point error; I'm not sure why this was happening.
